LibraryApp/src/components/books-list.js

```diff
@@ -38,13 +46,45 @@ class BooksList extends React.Component {
       isDataFetching: false,
       isRefreshing: false,
       searchText: '',
+      exitApp: 0,
     };
+    this.handleBackButtonClick = this.handleBackButtonClick.bind(this);
+    this.setExitApp = this.setExitApp.bind(this);
     this.onRefresh = this.onRefresh.bind(this);
     this.selectBook = this.selectBook.bind(this);
     this.search = this.search.bind(this);
     this.addNewBook = this.addNewBook.bind(this);
   }
 
+  componentDidMount() {
```

Comment:
Is it possible to take out all logic regarding back button feature to another component? books-list component should only be responsible for listing books and nothing else

Reply:
Also make sure that you use functional component with useState hook

Reply:
Done